app/resources/admin.html.template

```diff
@@ -68,38 +68,97 @@
          null; dropdown_index++) {
-      var language_code = $('language_selection_' + dropdown_index).value;
-      var language_title = $('language_title_' + dropdown_index).value;
+      var language_code = $('repo_titles_language_selection_' + 
+                             dropdown_index).value;
+      if (language_code_to_index_set[language_code] == 1) {
+        used_languages.splice(i-1, 1)
+      }
       if (language_code === 'unused') {
         continue;
       }
-      if (arr_contains(used_languages, language_code)) {
-        alert("You entered the same language (" + language_code + ") twice. " +
-              "That's not allowed.");
-        return false;
-      }
+      language_code_to_index_set[language_code] = 1;
       used_languages.push(language_code);
-      languages_to_titles[language_code] = language_title;
     }
 
     // Copy the dropdowns and the titles into the text boxes that will
     // actually be POSTed (language_menu_options and repo_titles).
     $('language_menu_options').value = JSON.stringify(used_languages);
-    $('repo_titles').value = JSON.stringify(languages_to_titles);
-
+    var index = 1;
+    for (var used_language in language_code_to_index_set) {
+      var language_value = $('repo_titles_language_value_' + index++).value
+      language_code_to_index_set[used_language] = language_value;
+    }
+    $('repo_titles').value = JSON.stringify(language_code_to_index_set);
+
+    for (var id_num = 0; id_num < MULTILINGUAL_TABLE_IDS.length; id_num++) {
+      var table_id = MULTILINGUAL_TABLE_IDS[id_num];
+      index = 1;
+      for(var used_language in language_code_to_index_set){
+        var language_value = $(table_id + '_language_value_' + index++).value;
+        language_code_to_index_set[used_language] = language_value;
+      }
+      $(table_id).value = JSON.stringify(language_code_to_index_set);
+    }
     return true;
   }
 
+  function get_selected_language(row_number) {
+    var selected_language_codes = 
+        $('repo_titles_language_selection_' + row_number);
+    var selected_index = selected_language_codes.selectedIndex;
+    var selected_language = selected_language_codes[selected_index].value;
+    return selected_language;
+  }
+
+  function validate_selected_languages(row_number) {
+    var used_languages = {};
+    var language_num = $('repo_titles_table').rows.length;
+    var selected_language = get_selected_language(row_number);
+    for (var index = 1; index < language_num; index++) {
+      var language_code = get_selected_language(index);
+      if (language_code != 'unused' && used_languages[language_code] != null) {
+        alert('You entered the same language!!');
+      }
+      used_languages[language_code] = 1;
+    }
+  }
+
+  function escape_html(str) {
+    var escape_table = {
+      '<': '&lt',
+      '>': '&gt',
+      '&': '&amp',
+      '"': '&quot',
+      '\'': '&#x27',
+    };
+    escaped_str = str.replace(/[<>&"\']/g, function (match) {
```

Comment:
Must be "var escaped_str", or you can just return it directly here.
Without "var", escaped_str will be a global variable.